### PR TITLE
PROJQUAY-70 - default wildcard mirror tags

### PR DESCRIPTION
**Pull request by thomasmckay**
https://issues.redhat.com/browse/PROJQUAY-70

When no mirror tag pattern is specified (empty string), default to wildcard (*). Additional clean up of mirror tag list to strip whitespace and remove duplicates.

### CODE REVIEW

File: data/model/repo_mirror.py [hunk 1]
```diff
@@ -527,6 +527,23 @@ def validate_rule(rule_type, rule_value):
         )
 
 
+def clean_rule(rule_type, rule_value):
```
Comment (kurtismullins) [inline data/model/repo_mirror.py]: I'm not opposed to doing this but we should also do validation and input sanitation on the frontend (the endpoint). It would be nice to enforce the input we receive from API Requests and give the appropriate response instead of silently mutating their data.

File: data/model/test/test_repo_mirroring.py [hunk 2]
```diff
@@ -245,3 +248,10 @@ def test_release_mirror(initialized_db):
     mirror = release_mirror(mirror, RepoMirrorStatus.FAIL)
     assert mirror.sync_retries_remaining == 3
     assert mirror.sync_start_date > original_sync_start_date
+
+
+@pytest.mark.parametrize(
+    "rule_value, cleaned_rule_value", [(["", " 1.1 ", "", "1.1"], ["*", "1.1"])]
```
Comment (kurtismullins) [inline data/model/test/test_repo_mirroring.py]: Can you also add the following test case?
```
"1.1, latest, "
```
(Note the trailing comma and extra space)

It *shouldn't* happen with JSON but a quick search reveals that *some* browsers will allow it. I am curious if, as an edge-case, we will accidentally inject `*` into a list of other tags.

File: data/model/repo_mirror.py (hunk 1 above)
Comment (alecmerdler) [reply data/model/repo_mirror.py]: Agreed, we should always avoid "magical" behavior whenever possible. Moving validation to the API layer and adding helpful tips in the frontend keeps business logic out of our data layer.

File: data/model/test/test_repo_mirroring.py (hunk 2 above)
Comment (thomasmckay) [reply data/model/test/test_repo_mirroring.py]: that's not how it works: a list is expected. if that comes in as a string that's what will be saved.
